proto/asn1/common/common.cc

```diff
@@ -0,0 +1,56 @@
+#include "common.h"
+#include <iostream>
+
+uint8_t GetVariableIntLen(const uint64_t value, const size_t base) {
+  uint8_t base_bits = log2(base);
+  for (uint8_t num_bits = (sizeof(value) - 1) * CHAR_BIT; num_bits >= base_bits;
+       num_bits -= base_bits) {
+    if (value >> num_bits) {
+      return ceil(static_cast<double>(num_bits) / base_bits) + 1;
+    }
+  }
+  // Special-case: zero requires one, not zero bytes.
+  return 1;
+}
+
+void InsertVariableInt(const size_t value,
+                       const size_t pos,
+                       std::vector<uint8_t>& der) {
+  std::vector<uint8_t> variable_int;
+  for (uint8_t shift = GetVariableIntLen(value, 256); shift != 0; --shift) {
+    variable_int.push_back((value >> ((shift - 1) * CHAR_BIT)) & 0xFF);
+  }
+  der.insert(der.begin() + pos, variable_int.begin(), variable_int.end());
+}
+
+void EncodeTagAndLength(const size_t tag,
```

Comment:
DESIGN: I think one of my concerns is that this now duplicates functionality: there are two ways to encode a tag here, using the `EncodeIdentifier` function (which handles all the tag variants) and this new function, which only handles 1-byte tags.

In general, we want to try to avoid reimplementing things when possible (although there are times where re-implementing is correct), and generally want to try to figure out "What's the right API shape here".

Let's discuss this in our 1:1 tomorrow for more about the design elements. This doesn't require an immediate change to this PR, other than fixing it up to take a `uint8_t tag_byte` rather than tag. I want to make sure we have a chance to discuss first.